test/sizeofchar.c

```diff
@@ -0,0 +1,41 @@
+#include "tests.h"
+#include <ctype.h>
+#include <wchar.h>
+
+int main(int argc, char **argv)
+{
+     int c, error = 0;
+
+     (void) argc; /* unused */
+     (void) argv; /* unused */
+
+     /* some simple sanity tests of  */
+     for (c = 0; c < 0x80; c++) {
```

Comment:
It would be good to check consistency with `utf8proc_encode_char`, i.e. check `utf8proc_sizeof_char(c) == utf8proc_encode_char(c, buf)` where you declare `utf8proc_uint8_t buf[4]`.